librz/arch/isa/x86/il_ops.inc

```diff
@@ -1595,19 +1596,21 @@ IL_LIFTER(popfq) {
 	return SEQ2(x86_il_set_flags(pop.val, 64), pop.eff);
 }
 
+#define CALCULATE_REX_PREFIX(rex) ((rex).W << 3 | (rex).R << 2 | (rex).X << 1 | (rex).B)
+
 RzILOpEffect *x86_push_helper_impl(RzILOpPure *val, unsigned int user_op_size, unsigned int bitness, const X86ILIns *ins, ut64 pc) {
 	unsigned int dflag = user_op_size;
 	unsigned int op_size;
 	unsigned int stack_size = bitness / BITS_PER_BYTE;
 
 	if (ins) {
 		if (bitness == 64) {
-			dflag = ins->structure->rex ? 8 : ins->structure->prefix[2] ? 2
-										    : 4; /* in bytes */
+			dflag = CALCULATE_REX_PREFIX(ins->structure->raw.rex) ? 8 : ins->structure->raw.prefixes[2].value ? 2
+															  : 4; /* in bytes */
```

Comment:
Move the `in bytes` comment to the define please.

Reply:
Yes, it seems to me that only `REX.W` needs to be checked. At the time, I might have misunderstood the comment in Capstone's source:

```cpp
/// REX prefix: only a non-zero value is relevant for x86_64
uint8_t rex;
```

Reply:
Lol, yeah. This is weird.